app/routes/_layout+/calendar.tsx

```diff
@@ -0,0 +1,180 @@
+import { Button } from "~/components/shared/button";
+import { ButtonGroup } from "~/components/shared/button-group";
+import { Spinner } from "~/components/shared/spinner";
+import calendarStyles from "~/styles/layout/calendar.css?url";
+import { appendToMetaTitle } from "~/utils/append-to-meta-title";
+import { getStatusClass } from "~/utils/calendar";
+import { makeShelfError } from "~/utils/error";
+import { data, error } from "~/utils/http.server";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.validator.server";
+import { requirePermission } from "~/utils/roles.server";
+
+export function links() {
+  return [{ rel: "stylesheet", href: calendarStyles }];
+}
+
+export const handle = {
+  breadcrumb: () => <Link to="/calendar">Calendar</Link>,
+};
+
+// Loader Function to Return Bookings Data
+export const loader = async ({ request, context }: LoaderFunctionArgs) => {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  try {
+    // @TODO here we have to handle self-service, and make sure they can only see bookings that belong to them
+    await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.booking,
+      action: PermissionAction.read,
+    });
+
+    const header = {
+      title: `Calendar`,
+    };
+
+    return json(data({ header }));
+  } catch (cause) {
+    const reason = makeShelfError(cause);
+    throw json(error(reason), { status: reason.status });
+  }
+};
+export const meta: MetaFunction<typeof loader> = ({ data }) => [
+  { title: appendToMetaTitle(data?.header.title) },
+];
+
+// Calendar Component
+const Calendar = () => {
+  const [error, setError] = useState<string | null>(null);
+  const [title, setTitle] = useState("");
+  const calendarRef = useRef<FullCalendar>(null);
+  const ripple = useRef<HTMLDivElement>(null);
+
+  const handleNavigation = (navigateTo: any) => {
+    const calendarApi = calendarRef.current?.getApi();
+    if (navigateTo == "prev") {
+      calendarApi?.prev();
+    } else if (navigateTo == "next") {
+      calendarApi?.next();
+    } else if (navigateTo == "today") {
+      calendarApi?.gotoDate(new Date());
+    }
+    updateTitle();
+  };
+
+  const updateTitle = () => {
+    const calendarApi = calendarRef.current?.getApi();
+    if (calendarApi) {
+      const currentDate = calendarApi.getDate();
+      const currentMonth = currentDate.toLocaleString("default", {
+        month: "long",
+      });
+      const currentYear = currentDate.getFullYear();
+      setTitle(`${currentMonth} ${currentYear}`);
+    }
+  };
+
+  const toggleSpinner = useCallback(
+    (state: any) => {
+      if (ripple.current) {
+        if (state) {
+          ripple.current.classList.remove("hidden");
+        } else {
+          ripple.current.classList.add("hidden");
+        }
+      }
+    },
+    [ripple]
+  );
+
+  useEffect(() => {
+    updateTitle();
```

Comment:
This is not needed. You can get the default title from the loader, for the first load.

Reply:
Fixed!